src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -63,7 +63,7 @@ SendCommandToWorker(char *nodeName, int32 nodePort, char *command)
 void
 SendCommandToFirstWorker(char *command)
 {
-	List *workerNodeList = ActivePrimaryNodeList();
+	List *workerNodeList = ActivePrimaryNodeList(NoLock);
```

Comment:
I believe this function is used when MX nodes need to agree on something? It seems it'd be better to have a singleton table `pg_dist_mx_lock_node` referencing a nodeid. Then one node is elected as the "first node" by looking up the value of that table. Then locks can be on that table when it comes to needing a consistent elected node